modules/local/locidex_select.nf

```diff
@@ -138,57 +140,86 @@ def select_locidex_db_path(db_values, db_name){
+        }
+
+        dates.add(date_check)
         if(date_check > max_date){
             max_date = date_check
-            max_date_idx = idx
+            max_date_entry = db_entry
         }
-        idx += 1
     }
 
     def max_date_count = dates.count(max_date)
     if(max_date_count > 1){
-        exit 1, "There are multiple versions of the most recent database for ${db_name}. Mikrokondo could not determine the best database to pick."
+        error("There are multiple versions of the most recent database for ${db_name}. Mikrokondo could not determine the best database to pick.")
+    }else if (max_date_count == 0){
+        error("There are not databases created after the year ${default_date}. Please set the allele database parameter, or adjust the date your database was created in the 'config.json'")
     }
-    return db_values[max_date_idx]
+
+    return max_date_entry
 }
 
 
-def tokenize_values(species, match_size){
-    // Tokenize the species values to find the right match
-    def tokens = []
+def window_string(species, match_size){
+    /*
+        Create an array of strings of a various match "match size" for comparison to a given value later one.
+
+        e.g. spieces is an array of: ["1", "2", "3", "4"] and match_size is 2 the output will be.
+            [
+                ["1", "2"],
+                ["2", "3"],
+                ["3", "4"]
+            ]
+    */
+    def tiles = []
     def adj_match_size = match_size - 1
     for(int spot = 0; spot < species.size()-adj_match_size; spot = spot + 1){
-        tokens.add(species[spot..spot + adj_match_size])
+        tiles.add(species[spot..spot + adj_match_size])
     }
-    return tokens
+    return tiles
 }
 
-def compare_lists(db_tokens, species_tokens){
-    // compare the various tokens till the right db is found
-    //! This allows for matches on shorter than optimal matches
-    for(i in species_tokens){
-        if(i == db_tokens){
-            return true
+def compare_lists(db_string, species_tokens){
+    /* compare the various windows till the right db is found
+        The db_string is an array of tokenized db values e.g. ["1", "2"] and the species would be tokenized into
+        would be [["1", "2"], ["2", "3"]], it would search through the windows of the spcies to see what matches
+        the database value best e.g. ["1", "2"].
+
+        This would match as true, however this allows for multiple matches to be found with similarly named databases.
+
+        To get a "better match" we will create a simple score of the the "match size" / length of the the db string tokens
+    */
+
+    for(window in species_tokens){
+        if(window == db_string){
+            // Can return on first match as any subsequent match would have the same score
+            def match_val = window.size() / db_string.size().toFloat()
+            return match_val
```

Comment:
Consider the array slice language as above.

What causes the the lengths of the compared elements to be different?

Reply:
fixed in: [4883b39](https://github.com/phac-nml/mikrokondo/pull/96/commits/4883b39c98601dc071a971c765b978f622a19a1c)